v-next/core/src/types/common.ts

```diff
@@ -57,11 +63,19 @@ const parameterTypeValidators: Record<
  *
  * This function uses a map of validators, where each validator is a function
  * that checks if a value is valid for a specific parameter type.
+ * If the parameter type is variadic, the value is considered valid if it is an
```

Comment:
I think we already discussed this, but `src/types` should only have types, not functions nor classes.

Reply:
Same for `parameterTypeValidators`, that's a value, not a type

Reply:
Yes, it's under a different issue. I'll do it once I've finished with the `run` method. See https://github.com/NomicFoundation/hardhat/issues/5304